### ISSUE TITLE
npm install failed because missing some files

### ISSUE THREAD

**Issue by codehz** [rprichard/winpty#55]
Although I do not think this problem is because of this reason...
```
deps\winpty\agent\AgentAssert.cc
deps\winpty\agent\winpty_wcsnlen.cc
deps\winpty\agent\winpty_wcsnlen.h
deps\winpty\agent\AgentAssert.h
```
When installing I use `npm install --msvs_version=2015`.

**Comment by rprichard** [comment]
I moved some files recently, and I notice that I forgot to update the gyp
file.  Maybe that's the reason?

I'll try to get the gyp file fixed soon, today if possible.

I guess npm is using master, probably because I haven't tagged a stable
release lately?

**Comment by rprichard** [comment]
I noticed that gyp complained about those missing files.  I pushed a commit that I think fixes the gyp issue, and it definitely fixes the missing file errors.  I can't test whether it fixes this issue yet, because I'm on vacation and don't have access to a decent Windows VM.

**Comment by codehz** [comment]
Thanks, This problem has been solved, but there is a question to ask (it may not be the issue of this project, so I don't create a new issue):
```
winpty_open_use_own_datapipe
winpty_exit
```
Are they the projects' obsolete API?[pty.js](https://github.com/chjj/pty.js) not build, because of them.

**Comment by rprichard** [comment]
I think pty.js is using a fork of winpty with those APIs added.  If you follow the winpty submodule link at https://github.com/chjj/pty.js/tree/master/deps, it leads to https://github.com/peters/winpty/tree/25e3186e9eb792d21ac02d69797a1aefc64c45d9.

**Comment by rprichard** [comment]
Good to hear the issue is fixed.  I'll close this issue then.

**Comment by rprichard** [comment]
@codehz I guess you're using my version of winpty somehow rather than the pty.js version?

I tried running `npm install pty --msvs_version=2015`, and it fails due to a few minor bugs in the pty.js fork that have been fixed in my version, e.g.:
 * A couple of files use `std::min` and `std::max` but fail to include `algorithm`.
 * As of C++11, if a string literal macro is concatenated to a string literal, there must be a separating space.  e.g. This doesn't work: `m_output->write(CSI"1;1H"CSI"2J");`
